utils/reposnapshot/snapshotmanager_test.go

```diff
@@ -192,17 +192,24 @@ func TestAddChildNode(t *testing.T) {
 	// Add child with no children pool.
 	addAndAssertChild(t, nil, root, CreateNewNode("no-pool", root))
 	// Add child with empty children pool.
-	pool := make(map[string]*Node)
-	addAndAssertChild(t, pool, root, CreateNewNode("empty-pool", root))
+	addAndAssertChild(t, []*Node{}, root, CreateNewNode("empty-pool", root))
 	// Add child with pool.
 	exists := CreateNewNode("exists", root)
-	pool[exists.name] = exists
-	addAndAssertChild(t, pool, root, exists)
+	addAndAssertChild(t, []*Node{exists}, root, exists)
 }
 
-func addAndAssertChild(t *testing.T, childrenMapPool map[string]*Node, root, expectedChild *Node) {
-	assert.NoError(t, root.AddChildNode(expectedChild.name, childrenMapPool))
-	assert.Equal(t, expectedChild, root.children[expectedChild.name])
+func addAndAssertChild(t *testing.T, childrenPool []*Node, root, expectedChild *Node) {
+	assert.NoError(t, root.AddChildNode(expectedChild.name, childrenPool))
+	assert.Equal(t, expectedChild, getChild(root, expectedChild.name))
+}
+
+func getChild(node *Node, childName string) *Node {
+	for i := range node.children {
```

Comment:
Why not Go style loop?
```suggestion
	for _, child := range node.children {
```